Src/BootCamp.Chapter/Inventory.cs

```diff
@@ -15,12 +16,75 @@ public Inventory()
 
         public Item[] GetItems(string name)
         {
-            return new Item[0];
+            var foundItems = new Item[0];
+            foreach (var item in _items)
+            {
+                if (item.GetName() == name)
+                {
+                    foundItems = Add(item);
+                }
+            }
+
+            return foundItems;
         }
 
         public void AddItem(Item item)
         {
+            _items = Add(item);
+        }
+
+        public Item[] Add(Item item)
+        {
+            var newArray = new Item[_items.Length + 1];
+            for (int i = 0; i <= _items.Length - 1; i++)
+            {
+                newArray[i] = _items[i];
+            }
+            newArray[^1] = item;
+            return newArray;
+        }
 
+        public bool SearchItem(string name)
+        {
+            foreach (var item in _items)
+            {
+                if (item.GetName() == name)
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        public void RemoveByName(string name)
```

Comment:
Hmm, can I remove things and return the old array. I understand from the lesson that you need a new array and return that one.  So I do not see how it can be void then.

Reply:
You need a new array, yes, but that doesn't mean that you need 2 methods.

Reply:
hmm, now  I wonder how I can use a new array and just returned void. 
I will sleep about it , Right now I do not see it. 

Lets say we have a array of 3 and we want to remove one item. 
Then we need to make a new array of 2 and copy the items we do not want to remove to the new array. 
But then we did not alter the old one. 

Or we use the old one  and changed the removed item to null. 
But that does not feel right because when you add a item then you get for example this  new array[] {null, "sword", null}

Reply:
had a idea. Hope it's allright this way